skyportal/tests/api/test_shifts.py

```diff
@@ -39,3 +41,129 @@ def test_shift(public_group, super_admin_token, view_only_token, super_admin_use
             for shift in data['data']
         ]
     )
+
+
+def test_shift_summary(
+    public_group,
+    super_admin_token,
+    super_admin_user,
+    upload_data_token,
+    view_only_token,
+    ztf_camera,
+):
+    # add a shift to the group, with a start day one day before today, and an end day one day after today
+    shift_name_1 = str(uuid.uuid4())
+    start_date = "2018-01-15T12:00:00"
+    end_date = "2018-01-17T12:00:00"
+    request_data = {
+        'name': shift_name_1,
+        'group_id': public_group.id,
+        'start_date': start_date,
+        'end_date': end_date,
+        'description': 'Shift during GCN',
+        'shift_admins': [super_admin_user.id],
+    }
+
+    status, data = api('POST', 'shifts', data=request_data, token=super_admin_token)
+    assert status == 200
+    assert data['status'] == 'success'
+
+    shift_id = data['data']['id']
+
+    shift_name_2 = str(uuid.uuid4())
+    start_date = "2018-01-17T12:00:00"
+    end_date = "2018-01-18T12:00:00"
+    request_data = {
+        'name': shift_name_2,
+        'group_id': public_group.id,
+        'start_date': start_date,
+        'end_date': end_date,
+        'description': 'Shift not during GCN',
+        'shift_admins': [super_admin_user.id],
+    }
+
+    status, data = api('POST', 'shifts', data=request_data, token=super_admin_token)
+    assert status == 200
+    assert data['status'] == 'success'
+
+    shift_id_2 = data['data']['id']
+
+    status, data = api('GET', f'shifts/{public_group.id}', token=super_admin_token)
+    assert status == 200
+    assert data['status'] == 'success'
+
+    datafile = f'{os.path.dirname(__file__)}/../data/GRB180116A_Fermi_GBM_Gnd_Pos.xml'
+    with open(datafile, 'rb') as fid:
+        payload = fid.read()
+    data = {'xml': payload}
+
+    status, data = api('POST', 'gcn_event', data=data, token=super_admin_token)
+    assert status == 200
+    assert data['status'] == 'success'
+
+    # wait for event to load
+    time.sleep(15)
```

Comment:
Done !

Reply:
I would do a loop to try the API, check for success, and if the loop times out then raise an error. 

If you do a loop it can be 30 seconds or more, as long as most of the time the test runs in only a few seconds. 

We should also be working on smaller datasets, as we discussed last week with @mcoughlin